cpp/models/ide_secir/model.cpp

```diff
@@ -124,79 +141,128 @@ void Model::initial_compute_compartments(ScalarType dt)
+    for (int group = 0; group < m_num_agegroups; ++group) {
+        int Si = get_state_flat_index(Eigen::Index(InfectionState::Susceptible), group);
+        if (m_populations[Eigen::Index(0)][Si] < 1e-12) {
+            Susceptibles_given = false;
+            break;
+        }
+    }
+    bool Recovered_given = true;
+    for (int group = 0; group < m_num_agegroups; ++group) {
+        int Ri = get_state_flat_index(Eigen::Index(InfectionState::Recovered), group);
+        if (m_populations[Eigen::Index(0)][Ri] < 1e-12) {
+            Recovered_given = false;
+            break;
+        }
+    }
 
-        // The scheme of the ODE model for initialization is applied here.
-        m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Recovered)] =
-            m_total_confirmed_cases - m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedSymptoms)] -
-            m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedSevere)] -
-            m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedCritical)] -
-            m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Dead)];
-
-        m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Susceptible)] =
-            m_N - m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Exposed)] -
-            m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedNoSymptoms)] -
-            m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedSymptoms)] -
-            m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedSevere)] -
-            m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedCritical)] -
-            m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Recovered)] -
-            m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Dead)];
+    // We check which Initialization Method we want to use.
+    if (!m_total_confirmed_cases.empty() &&
+        std::all_of(m_total_confirmed_cases.begin(), m_total_confirmed_cases.end(), [](ScalarType x) {
+            return x > 1e-12;
+        })) {
+        m_initialization_method = 1;
+        for (int group = 0; group < m_num_agegroups; ++group) {
+            int Si = get_state_flat_index(Eigen::Index(InfectionState::Susceptible), group);
+            int Ei = get_state_flat_index(Eigen::Index(InfectionState::Exposed), group);
+            int Ci = get_state_flat_index(Eigen::Index(InfectionState::InfectedNoSymptoms), group);
+            int Ii = get_state_flat_index(Eigen::Index(InfectionState::InfectedSymptoms), group);
+            int Hi = get_state_flat_index(Eigen::Index(InfectionState::InfectedSevere), group);
+            int Ui = get_state_flat_index(Eigen::Index(InfectionState::InfectedCritical), group);
+            int Ri = get_state_flat_index(Eigen::Index(InfectionState::Recovered), group);
+            int Di = get_state_flat_index(Eigen::Index(InfectionState::Dead), group);
```

Comment:
```suggestion
            int Si = get_state_flat_index(Eigen::Index(InfectionState::Susceptible), group);
            int Ei = get_state_flat_index(Eigen::Index(InfectionState::Exposed), group);
            int INSi = get_state_flat_index(Eigen::Index(InfectionState::InfectedNoSymptoms), group);
            int ISyi = get_state_flat_index(Eigen::Index(InfectionState::InfectedSymptoms), group);
            int ISevi = get_state_flat_index(Eigen::Index(InfectionState::InfectedSevere), group);
            int ICri = get_state_flat_index(Eigen::Index(InfectionState::InfectedCritical), group);
            int Ri = get_state_flat_index(Eigen::Index(InfectionState::Recovered), group);
            int Di = get_state_flat_index(Eigen::Index(InfectionState::Dead), group);
```
In the code we use InfectedNoSymptoms etc. for the infection states which is why I would use according abbreviations.